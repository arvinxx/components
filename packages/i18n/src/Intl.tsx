import React from 'react';
import { IntlProvider as IntlProvider_ } from 'react-intl';
import type { IntlProviderProps } from './type';

export const IntlProvider: IntlProviderProps = (props) => (
  <IntlProvider_ {...props} />
);
