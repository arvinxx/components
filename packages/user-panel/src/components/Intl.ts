import { useTranslation } from 'react-i18next';

export const useFormatMessage = () => {
  const { t } = useTranslation();

  return (id, values?) => t(id, values);
};
