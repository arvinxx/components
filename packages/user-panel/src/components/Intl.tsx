import React from 'react';
import { IntlProvider as IntlProvider_ } from 'react-intl';
import type { LocaleMessages } from '../type';

export const IntlProvider: React.FC<
  Omit<React.ComponentProps<typeof IntlProvider_>, 'messages'> & {
    messages: LocaleMessages;
  }
> = (props) => <IntlProvider_ {...props} />;
