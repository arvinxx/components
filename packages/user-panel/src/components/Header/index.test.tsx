import React from 'react';
import { render } from '@testing-library/react';

import { Header } from './index';
import { IntlProvider } from '../Intl';

describe('Header 组件', () => {
  it('登录头部', () => {
    const { container } = render(
      <IntlProvider>
        <Header type={'login'} />
      </IntlProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
