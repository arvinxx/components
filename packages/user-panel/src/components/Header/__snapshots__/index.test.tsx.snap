// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 组件 登录头部 1`] = `
<div>
  <div
    class="avx-user-panel-header"
  >
    <h2
      class="ant-typography"
      style="text-align: center; margin-bottom: 24px; font-weight: normal;"
    >
      Login
    </h2>
  </div>
</div>
`;
